reVX/least_cost_xmission/aoswt_utilities.py

```diff
@@ -0,0 +1,604 @@
+"""
+Various utility functions to prep data for AOSWT processing.
+
+Mike Bannister 5/2022
+"""
+import os
+import json
+import h5py
+import numpy as np
+import pandas as pd
+import geopandas as gpd
+import rasterio as rio
+from functools import reduce
+from rasterio import features
+from rasterio.warp import reproject, Resampling
+from shapely.geometry import Point, LineString
+
+import rex
+
+
+def _sum(a, b):
+    return a + b
+
+
+def convert_pois_to_lines(poi_csv_f, template_f, out_f):
+    """
+    Convert POIs in CSV to lines and save in a geopackage as substations. Also
+    create a fake transmission line to connect to the substations.
+
+    Parameters
+    ----------
+    poi_csv_f : str
+        Path to CSV file with POIs in it
+    template_f : str
+        Path to template raster with CRS to use for geopackage
+    out_f : str
+        Path and file name for geopackage
+    """
+    print(f'Converting POIs in {poi_csv_f} to lines in {out_f}')
+    with rio.open(template_f) as ras:
+        crs = ras.crs
+
+    df = pd.read_csv(poi_csv_f)[['POI Name', 'State', 'Voltage (kV)', 'Lat',
+                                 'Long']]
+
+    pts = gpd.GeoDataFrame(df, geometry=gpd.points_from_xy(df.Long, df.Lat))
+    pts = pts.set_crs('EPSG:4326')
+    pts = pts.to_crs(crs)
+
+    # Convert points to short lines
+    new_geom = []
+    for pt in pts.geometry:
+        end = Point(pt.x + 50, pt.y + 50)
+        line = LineString([pt, end])
+        new_geom.append(line)
+    lines = pts.set_geometry(new_geom, crs=crs)
+
+    # Append some fake values to make the LCP code happy
+    lines['ac_cap'] = 9999999
+    lines['category'] = 'Substation'
+    lines['voltage'] = 69
```

Comment:
maybe make the fake voltage 0 or 1e9?

Reply:
The substation needs a transmission line with adequate voltage for the code to connect to it. I arbitrarily chose 69kV as a standard voltage. 1e9 may work as well but I would need to check it.